qadence/mitigations/readout.py

```diff
@@ -25,51 +60,85 @@ def renormalize_counts(corrected_counts: npt.NDArray, n_shots: int) -> npt.NDArr
         corrected_counts -= counts_diff
         corrected_counts = np.where(corrected_counts < 0, 0, corrected_counts)
         sum_corrected_counts = sum(corrected_counts)
-        if sum_corrected_counts < n_shots:
-            renormalization_factor = max(sum_corrected_counts, n_shots) / min(
-                sum_corrected_counts, n_shots
-            )
-        else:
-            renormalization_factor = min(sum_corrected_counts, n_shots) / max(
-                sum_corrected_counts, n_shots
-            )
+
+        renormalization_factor = n_shots / sum_corrected_counts
         corrected_counts = np.rint(corrected_counts * renormalization_factor).astype(int)
     return corrected_counts
 
 
+def matrix_inv(K: npt.NDArray) -> npt.NDArray:
+    return inv(K) if matrix_rank(K) == K.shape[0] else pinv(K)
+
+
 def mitigation_minimization(
-    noise: Noise, mitigation: Mitigations, samples: list[Counter]
+    noise: Noise,
+    mitigation: Mitigations,
+    samples: list[Counter],
 ) -> list[Counter]:
     """Minimize a correction matrix subjected to stochasticity constraints.
 
     See Equation (5) in https://arxiv.org/pdf/2001.09980.pdf.
+    See Page(3) in https://arxiv.org/pdf/1106.5458.pdf for MLE implementation
+
+    Args:
+        noise: Specifies confusion matrix and default error probability
+        mitigation: Selects additional mitigation options based on noise choice.
+        For readout we have the following mitigation options for optimization
+        1.constrained_opt, 2.mle. Default : mle
```

Comment:
Change 1.constrained_opt, 2.mle. Default : mle  to 1. CONSTRAINED 2. MLE. Default : MLE